cpp/src/search/contains_table.cu

```diff
@@ -205,29 +205,56 @@ rmm::device_uvector<bool> contains_with_lists_or_nans(table_view const& haystack
       auto const row_bitmask_ptr        = bitmask_buffer_and_ptr.second;
 
       // Insert only rows that do not have any null at any level.
-      auto const insert_map = [&](auto const value_comp) {
-        auto const d_eqcomp = strong_index_comparator_adapter{
-          comparator.equal_to(nullate::DYNAMIC{haystack_has_nulls}, compare_nulls, value_comp)};
-        map.insert_if(haystack_it,
-                      haystack_it + haystack.num_rows(),
-                      thrust::counting_iterator<size_type>(0),  // stencil
-                      row_is_valid{row_bitmask_ptr},
-                      d_hasher,
-                      d_eqcomp,
-                      stream.value());
-      };
+      if (cudf::detail::has_nested_columns(haystack)) {
+        auto const insert_map = [&](auto const value_comp) {
+          auto const d_eqcomp = strong_index_comparator_adapter{comparator.equal_to<true>(
+            nullate::DYNAMIC{haystack_has_nulls}, compare_nulls, value_comp)};
+          map.insert_if(haystack_it,
+                        haystack_it + haystack.num_rows(),
+                        thrust::counting_iterator<size_type>(0),  // stencil
+                        row_is_valid{row_bitmask_ptr},
+                        d_hasher,
+                        d_eqcomp,
+                        stream.value());
+        };
+
+        dispatch_nan_comparator(compare_nans, insert_map);

```

Comment:
Check for `has_nested_columns` inside `insert_map` to reduce code duplicate. The lambda `insert_map` doesn't depend on `d_eqcomp`.

Reply:
It does depend on `d_eqcomp`, the line with `map.insert_if(..., d_eqcomp)` uses it